akka-durable-mailboxes/akka-file-mailbox/src/main/scala/akka/actor/mailbox/filebased/filequeue/Journal.scala

```diff
@@ -86,14 +86,15 @@ class Journal(queuePath: String, syncJournal: ⇒ Boolean, log: LoggingAdapter)
       add(false, item)
     }
     if (syncJournal) writer.force(false)
-    writer.close
-    tmpFile.renameTo(queueFile)
-    open
+    writer.close()
+    // as renameTo is not guaranteed to be able to overwrite the target file, delete it explicitly
+    try queueFile.delete() finally tmpFile.renameTo(queueFile)
```

Comment:
do we really need the finally here? Can we expect the rename to succeed if the delete fails?